httpx/_content.py

```diff
@@ -187,6 +198,7 @@ def encode_request(
     files: Optional[RequestFiles] = None,
     json: Optional[Any] = None,
     boundary: Optional[bytes] = None,
+    headers: Optional[Mapping[str, str]] = None,
```

Comment:
How about we keep things a little tighter here by passing `content_type: str = None` instead? That restricts the information we're passing around to the one thing we're actually interested in.

Reply:
Yup good call